applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -311,15 +314,20 @@ def get_mechanical_solution_strategy(self):
         return self._mechanical_solution_strategy
 
     def import_constitutive_laws(self):
-        materials_filename = self.settings["material_import_settings"]["materials_filename"].GetString()
-        if (materials_filename != ""):
-            # Add constitutive laws and material properties from json file to model parts.
-            material_settings = KratosMultiphysics.Parameters("""{"Parameters": {"materials_filename": ""}} """)
-            material_settings["Parameters"]["materials_filename"].SetString(materials_filename)
-            KratosMultiphysics.ReadMaterialsUtility(material_settings, self.model)
+        if self.settings["material_import_settings"].Has("custom_reader"): # We use our own file for reading
+            custom_reader = import_module(self.settings["material_import_settings"]["custom_reader"].GetString())
+            custom_reader.read(self.settings["material_import_settings"])
```

Comment:
```suggestion
            custom_reader.ReadMaterials(self.model, self.settings["material_import_settings"])
```

you forgot to pass the `Model`